### PR TITLE
Allow to change name of the integration

### PR DESCRIPTION
**Pull request by cashew22**
The patch allow the user to change the title of the integration. It also remove the default name of the integration based on the hostname and use Moonraker instead.

The patch also rework the async_reload_entry method to reload the configuration instead of unloading and resetuping the integration.

#27

### CODE REVIEW

**Comment by marcolivierarsenault** [comment]
image
This should be something else. (Not sure what)

**Comment by marcolivierarsenault** [comment]
image
Same thing here? I guess

**Comment by cashew22** [comment]
By using "DOMAIN", the title become "moonraker" instead of "Moonraker", that's why I added a new constant in the first place. I suggest that either we change DOMAIN to "Moonraker" but I don't know the implication of that or, we add some code to capitalize the "m" before setting the titlle.

File: custom_components/moonraker/config_flow.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ def __init__(self):
         """Initialize."""
         _LOGGER.debug("loading moonraker confFlowHandler")
         self._errors = {}
-        self.title = None
+        self.title = DOMAIN
```
Comment (cashew22) [inline custom_components/moonraker/config_flow.py:21]: I think we can just set it directly in async_create_entry

Comment (marcolivierarsenault) [reply custom_components/moonraker/config_flow.py:21]: Ideally I want this to be the hostname (can be next release)

**Comment by marcolivierarsenault** [comment]
>By using "DOMAIN", the title become "moonraker" instead of "Moonraker", that's why I added a new constant in the first place. I suggest that either we change DOMAIN to "Moonraker" but I don't know the implication of that or, we add some code to capitalize the "m" before setting the titlle.

I want to move to the hostname, so I would keep if for Domain for now. Easier swap

**Review by marcolivierarsenault — APPROVED** [review]

**Pull request merged**